packages/medusa/src/models/category.ts

```diff
@@ -0,0 +1,112 @@
+import { generateEntityId } from "../utils/generate-entity-id"
+import { SoftDeletableEntity } from "../interfaces/models/soft-deletable-entity"
+import {
+  BeforeInsert,
+  Index,
+  Entity,
+  Tree,
+  Column,
+  PrimaryGeneratedColumn,
+  TreeChildren,
+  TreeParent,
+  TreeLevelColumn,
+  JoinColumn,
+} from "typeorm"
+
+@Entity()
+@Tree("materialized-path")
+export class Category extends SoftDeletableEntity {
```

Comment:
suggestion: Rename to `ProductCategory` to not clash with future categories in other domain

Reply:
And be consistent with how types, collection, and tags are named

Reply:
might want to do the same for the table?

Reply:
Good point, renaming.